src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -14,10 +14,137 @@
+import codeu.model.data.User;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
-/** Servlet class responsible for the chat page. */
+/** Servlet class responsible for the Admin page. */
 public class AdminServlet extends HttpServlet {
-  // TODO(JW): To be implemented.
 
+  /** Store class that gives access to Conversations. */
+  private ConversationStore conversationStore;
+
+  /** Store class that gives access to Messages. */
+  private MessageStore messageStore;
+
+  /** Store class that gives access to Users. */
+  private UserStore userStore;
+
+  /** Set up state for handling data collection for the Admin Page. */
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setConversationStore(ConversationStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+    setUserStore(UserStore.getInstance());
+  }
+
+  /**
+   * Sets the ConversationStore used by this servlet. This function provides a common setup method
+   * for use by the test framework or the servlet's init() function.
+   */
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+
+  /**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method for
+   * use by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    request.setAttribute("totalUsers", userStore.getUsers().size());
+    request.setAttribute("totalConversations", conversationStore.getAllConversations().size());
+    request.setAttribute("totalMessages", messageStore.getTotalMessages());
+    request.setAttribute("mostActive", getMostActiveUser());
+    request.setAttribute("newestUser", getNewestUser());
+    request.setAttribute("wordiestUser", getWordiestUser());
+    request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    String username = (String) request.getSession().getAttribute("user");
+    Boolean admin = userStore.getUser(username).isAdmin();
+    if (!admin) {
```

Comment:
I think this check actually needs to be in `doGet`, so that it's checked when the user tries to access the page.

Also maybe throw a short unit test in for it.

Reply:
Oh, and make sure to check whether there is a valid logged in user, otherwise you'll get a NullPointerException because `getUser` will return null.

Reply:
But actually I think it would be fine if you just removed this part from doPost and merged this PR and then did the permissions thing in a new PR.  This one's gotten pretty long already.